storagemarket/logs/log.go

```diff
@@ -13,75 +13,84 @@ import (
 )
 
 type DealLogger struct {
-	ctx    context.Context
-	logger *logging.ZapEventLogger
-	logsDB *db.LogsDB
-	prefix string
+	ctx       context.Context
+	logger    *logging.ZapEventLogger
+	logsDB    *db.LogsDB
+	subsystem string
 }
 
 func NewDealLogger(ctx context.Context, logsDB *db.LogsDB) *DealLogger {
-	prefix := "boost-storage-deal"
 	return &DealLogger{
 		ctx:    ctx,
-		logger: logging.Logger(prefix),
+		logger: logging.Logger("boost-storage-deal"),
 		logsDB: logsDB,
-		prefix: prefix,
 	}
 }
 
 func (d *DealLogger) Subsystem(name string) *DealLogger {
-	prefix := d.prefix + "/" + name
 	return &DealLogger{
-		ctx:    d.ctx,
-		logger: logging.Logger(prefix),
-		logsDB: d.logsDB,
-		prefix: prefix,
+		ctx:       d.ctx,
+		logger:    logging.Logger(d.subsystem + name),
+		logsDB:    d.logsDB,
+		subsystem: name,
 	}
 }
 
 func (d *DealLogger) Infow(dealId uuid.UUID, msg string, kvs ...interface{}) {
-	d.logger.Infow(msg, "id", dealId, kvs)
+	kvs = paramsWithDealID(dealId, kvs...)
+
+	d.logger.Infow(msg, kvs...)
 	d.updateLogDB(dealId, msg, "INFO", kvs)
 }
 
 func (d *DealLogger) Warnw(dealId uuid.UUID, msg string, kvs ...interface{}) {
-	d.logger.Warnw(msg, "id", dealId, kvs)
+	kvs = paramsWithDealID(dealId, kvs...)
+	d.logger.Warnw(msg, kvs...)
 	d.updateLogDB(dealId, msg, "WARN", kvs)
 }
 
 func (d *DealLogger) Errorw(dealId uuid.UUID, errMsg string, kvs ...interface{}) {
-	d.logger.Errorw(errMsg, "id", dealId, kvs)
+	kvs = paramsWithDealID(dealId, kvs...)
+
+	d.logger.Errorw(errMsg, kvs...)
 	d.updateLogDB(dealId, errMsg, "ERROR", kvs)
 }
 
-func (d *DealLogger) updateLogDB(dealId uuid.UUID, msg string, level string, kvs ...interface{}) {
-	jsn, err := json.Marshal(kvs)
-	if err != nil {
-		d.logger.Warnw("failed to marshal log params to json", "err", err, "id", dealId)
-	}
+func (d *DealLogger) LogError(dealId uuid.UUID, errMsg string, err error) {
+	d.logger.Errorw(errMsg, "id", dealId, "err", err)
 
 	l := &db.DealLog{
 		DealUUID:  dealId,
 		CreatedAt: time.Now(),
-		LogLevel:  level,
-		LogMsg:    d.prefix + ":" + msg,
-		LogParams: string(jsn),
+		LogLevel:  "ERROR",
+		LogMsg:    fmt.Sprintf("msg: %s, err: %s", errMsg, err),
```

Comment:
For the LogError method why not just call Errorw:
```
d.Errorw(dealId, errMsg, "err", err)
```

Reply:
Done